ssm/twarhmm/models.py

```diff
@@ -0,0 +1,176 @@
+                0.1 / (num_discrete_states - 1) * (1 - np.eye(num_discrete_states))
+        else:
+            assert discrete_state_transition_matrix.shape == (num_discrete_states, num_discrete_states)
+            assert np.allclose(np.sum(discrete_state_transition_matrix, axis=1), 1.0)
+        discrete_state_transitions = \
+                StationaryTransitions(num_discrete_states, discrete_state_transition_matrix)
+
+        # Initialize the time constant transitions
+        time_constant_transitions = \
+            SimpleStickyTransitions(num_time_constants, time_constant_stay_probability)
+
+        transitions = FactorialTransitions((discrete_state_transitions, time_constant_transitions))
+
+        # Initialize the emissions
+        if emission_weights is None:
+            assert seed is not None and num_emission_dims is not None, \
+                "You must either specify the emission_weights or give a dimension "\
+                "and seed (PRNGKey) so that they can be initialized randomly."
+            this_seed, seed = jr.split(seed, 2)
+            emission_weights = tfd.Normal(0, 0.1).sample(
+                seed=this_seed,
+                sample_shape=(num_discrete_states, num_emission_dims, num_emission_dims))
+
+        if emission_biases is None:
+            assert seed is not None and num_emission_dims is not None, \
+                "You must either specify the emission_weights or give a dimension, "\
+                "number of lags, and seed (PRNGKey) so that they can be initialized randomly."
+            this_seed, seed = jr.split(seed, 2)
+            emission_biases = tfd.Normal(0, .1).sample(
+                seed=this_seed,
+                sample_shape=(num_discrete_states, num_emission_dims))
+
+        if emission_covariances is None:
+            assert num_emission_dims is not None, \
+                "You must either specify the emission_covariances or give a dimension "\
+                "so that they can be initialized."
+            emission_covariances = np.tile(np.eye(num_emission_dims), (num_discrete_states, 1, 1))
+
+        emissions = TimeWarpedAutoregressiveEmissions(
+            num_discrete_states,
+            time_constants,
+            weights=emission_weights,
+            biases=emission_biases,
+            scale_trils=np.linalg.cholesky(emission_covariances),
+            emissions_distribution_prior=emission_prior)
+
+        super().__init__(num_states, initial_condition, transitions, emissions)
+
+    @property
+    def num_discrete_states(self):
+        return self.num_states[0]
+
+    @property
+    def num_time_constants(self):
+        return self.num_states[1]
+
+    @property
+    def emission_weights(self):
+        return self._emissions._weights
+
+    @property
+    def emission_biases(self):
+        return self._emissions._biases
+
+    @property
+    def emission_scale_trils(self):
+        return self._emissions._scale_trils
+
+    @format_dataset
```

Comment:
I think we'll want to remove this decorator as it's no longer in the main branch.

Reply:
Fixed in https://github.com/lindermanlab/ssm-jax-refactor/pull/14/commits/3cac3bbd888e4d5e1b7d52faf4d7209efef8563f